server_test.go

```diff
@@ -763,6 +763,112 @@ func TestServerResponseBodyStream(t *testing.T) {
 	}
 }
 
+func TestServerResponseBodyStreamProxy(t *testing.T) {
+	t.Parallel()
+
+	ln := fasthttputil.NewInmemoryListener()
+	lnProxy := fasthttputil.NewInmemoryListener()
+
+	h := func(ctx *RequestCtx) {
+		ctx.SetConnectionClose()
+		body := []byte("body")
+		ctx.SetBodyStream(bytes.NewBuffer(body), len(body))
+	}
+
+	hProxy := func(ctx *RequestCtx) {
+		ctx.SetConnectionClose()
+
+		cli := Client{
+			Dial: func(addr string) (net.Conn, error) {
+				return lnProxy.Dial()
+			},
+		}
+		err := cli.Do(&ctx.Request, &ctx.Response)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		responseBodyStream := ctx.ResponseBodyStream()
+		ctx.Response.Body()
+		body, _ := io.ReadAll(responseBodyStream)
+		body = append(body, []byte(" append")...)
+		ctx.SetBodyStream(bytes.NewBuffer(body), len(body))
```

Comment:
Is it not possible to do `ctx.SetBodyStream(responseBodyStream, -1)` here? I feel like that's what people are going to want to do when we introduce this new function. So that the full body doesn't need to be read into memory anymore like you're still doing here.